swarm_copy_tests/tools/test_literature_search_tool.py

```diff
@@ -0,0 +1,49 @@
+"""Tests Literature Search tool."""
+
+from unittest.mock import AsyncMock, Mock
+
+import httpx
+import pytest
+
+from swarm_copy.tools import LiteratureSearchTool
+from swarm_copy.tools.literature_search_tool import ParagraphMetadata, LiteratureSearchMetadata, LiteratureSearchInput
+
+
+class TestLiteratureSearchTool:
+    @pytest.mark.asyncio
+    async def test_arun(self):
+        url = "http://fake_url"
+        reranker_k = 5
+
+        client = httpx.AsyncClient()
+        client.get = AsyncMock()
+        response = Mock()
+        response.status_code = 200
+        client.get.return_value = response
+        response.json.return_value = [
+            {
+                "article_title": "Article title",
+                "article_authors": ["Author1", "Author2"],
+                "paragraph": "This is the paragraph",
+                "section": "fake_section",
+                "article_doi": "fake_doi",
+                "journal_issn": "fake_journal_issn",
+            }
+            for _ in range(reranker_k)
+        ]
+
```

Comment:
Why not just use the `httpx_mock` fixture as in the other tests ?

Reply:
fixed